src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -158,14 +159,106 @@ class CsvToHttp extends ApiHandler {
     return { httpBody, convertionErrorBuffer }
   }
 
+  /**
+   * Get only the valid mapping fields
+   * @param {Mixed} allHeaders - available headears in the csvFile
+   * @return {Array} - Return a bool
+   */
+  getOnlyValidMappingValue(allHeaders) {
+    const onlyValidMappingValue = []
+    this.mapping.forEach((mapping) => {
+      if (!CsvToHttp.isHeaderValid(allHeaders, mapping.csvField)) {
+        this.logger.warn(`The header: '${mapping.csvField}' is not present in the csv file`)
+      } else {
+        onlyValidMappingValue.push(mapping)
+      }
+    })
+    return onlyValidMappingValue
+  }
+
+  /**
+   * @param {Array} httpBody - Body to send
+   * @return {Promise} - Promise
+   */
+  sendData(httpBody) {
+    return new Promise((resolve, reject) => {
+      try {
+        if (httpBody.length > this.bodyMaxLength) {
+          // Divide the current body in array of maximun maxLength elements
+          let i = 0
+          for (i; i < httpBody.length; i += this.bodyMaxLength) {
+            this.sendRequest(httpBody.slice(i, i + this.bodyMaxLength - 1))
+          }
+        } else {
+          this.sendRequest(httpBody)
+        }
+        resolve(true)
+      } catch (error) {
+        this.logger.error(error)
+        reject()
+      }
+    })
+  }
+
+  /**
+   * Send Request(s) to the selected host
+   * @param {Array} body - Body to send
+   * @return {void}
+   */
+  async sendRequest(body) {
+    // Create a base header in order not to send a request with a file but a content
+    const baseHeaders = { 'Content-Type': 'application/json' }
+    try {
+      await this.engine.requestService.httpSend(
+        this.request.host,
+        this.request.method,
+        this.request.authenticationField,
+        this.proxy,
+        JSON.stringify(body),
+        baseHeaders,
+      )
+    } catch (error) {
+      this.logger.error(error)
+    }
+  }
+
+  /**
+   * Test if the field is valid thanks to all headers
+   * @param {Mixed} allHeaders - available headears in the csvFile
+   * @param {Mixed} field - field to test if it matchs with the available headears
+   * @return {Boolean} - Return a bool
+   */
+  static isHeaderValid(allHeaders, field) {
+    if (field.match(REGEX_CONTAIN_VARIABLE_STRING)) {
+      const csvFieldSplit = field.split(REGEX_SPLIT_TEMPLATE_STRING).filter(Boolean)
+      csvFieldSplit.every((element) => {
```

Comment:
You must add "return" at the start of the line. Otherwise, this test (the 'every' loop) never returns true or false. it then goes on to return false on line 248 since the field is not found on the allHearders tab